### PR TITLE
Consistent package scripts and output

### LINKED ISSUE #5433: Consistent package scripts
- [ ] Make sure all packages have consistent output e.g. `dist`
- [ ] Make sure all packages have consistent script names
- [ ] Remove top level dependency on build and type-check
- [ ] Remove top level workspace scripts, add docs or find replacement

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Fixes #5433 

### WHAT is this pull request doing?

- [x] Make sure all packages have consistent output e.g. `dist`
- [x] Make sure all packages have consistent script names
- [x] Remove top level workspace scripts, add docs or find replacement
- [ ] ~~Remove top level dependency on build and type-check~~

### CODE REVIEW

File: polaris-for-vscode/server/src/server.ts [hunk 1]
```diff
@@ -62,28 +61,29 @@ connection.onInitialize((params: InitializeParams) => {
...
-            ? CompletionItemKind.Color
-            : CompletionItemKind.Variable,
-      };
-    });
-
-    groupedCompletionItems[category] = completionItems;
-
-    allCompletionItems = allCompletionItems.concat(completionItems);
+  for (const tokenGroup in groupedTokens) {
+    if (Object.prototype.hasOwnProperty.call(groupedTokens, tokenGroup)) {
```
Comment (aaronccasanova) [inline polaris-for-vscode/server/src/server.ts:65]:
```suggestion
    if (groupedTokens.hasOwnProperty(tokenGroup)) {
```

File: polaris-for-vscode/server/src/server.ts [hunk 2]
```diff
@@ -104,12 +104,18 @@ connection.onCompletion(
...
 
     for (const tokenGroup in tokenGroupPatterns) {
-      const category = tokenGroup as keyof typeof tokenGroupPatterns;
-
-      if (tokenGroupPatterns[category].test(currentText)) {
-        const currentCompletionItems = groupedCompletionItems[category];
-        if (currentCompletionItems) {
-          matchedCompletionItems = matchedCompletionItems.concat(currentCompletionItems);
+      if (
+        Object.prototype.hasOwnProperty.call(tokenGroupPatterns, tokenGroup)
```
Comment (aaronccasanova) [inline polaris-for-vscode/server/src/server.ts:108]: Same as above

File: .github/workflows/ci-stylelint-polaris.yml [hunk 3]
```diff
@@ -25,14 +25,11 @@ jobs:
...
           cache: yarn
       - uses: actions/cache@v2
-        name: Restore build cache
+        name: Restore cache
         with:
           path: |
-            .loom
+            .eslintcache
             **/*.tsbuildinfo
-            polaris-react/.loom
```
Comment (aaronccasanova) [inline .github/workflows/ci-stylelint-polaris.yml:33]: Should we keep this around until we remove the root level `yarn build` step from this workflow?

File: polaris-for-vscode/package.json [hunk 4]
```diff
@@ -18,36 +18,27 @@
     "polaris",
     "shopify"
   ],
-  "engines": {
-    "vscode": "^1.64.0"
-  },
```
Comment (aaronccasanova) [inline polaris-for-vscode/package.json]: This appears to be a required field: https://code.visualstudio.com/api/references/extension-manifest

File: .github/workflows/ci-stylelint-polaris.yml (hunk 3 above)
Comment (sam-b-rose) [reply .github/workflows/ci-stylelint-polaris.yml:33]: I think this is still safe to remove. From what I noticed, it seems the only the `polaris-react` test data from Jest was being stored here. So, I think we only need to cache this for the `polaris-react` CI workflow

File: polaris-for-vscode/package.json (hunk 4 above)
Comment (sam-b-rose) [reply polaris-for-vscode/package.json]: Ah, good catch! Yarn was complaining about this not being a valid engine type... will add back!

File: polaris-for-vscode/server/src/server.ts (hunk 1 above)
Comment (sam-b-rose) [reply polaris-for-vscode/server/src/server.ts:65]: I actually did try this initially but got [this ESLint error of `no-prototype-builtins`](https://eslint.org/docs/rules/no-prototype-builtins) :(

Comment (aaronccasanova) [reply polaris-for-vscode/server/src/server.ts:65]: Oof and `Object.hasOwn(groupedTokens, tokenGroup)` is only available in Node >=16.9.0

File: package.json [hunk 5]
```diff
@@ -32,6 +27,7 @@
     "@shopify/stylelint-plugin": "^11.0.0",
     "@shopify/typescript-configs": "^5.1.0",
     "eslint": "^8.3.0",
+    "npm-run-all": "^4.1.5",
```
Comment (alex-page) [inline package.json:30]: `npm-run-all` doesn't seem to be used 🤔

Comment (aaronccasanova) [reply package.json:30]: https://github.com/Shopify/polaris/blob/6ad54d32d0cedf507d7fdd66d2d5677ce0787098/polaris-for-vscode/package.json#L33-L35

Comment (sam-b-rose) [reply package.json:30]: It is being used for running all the lint scripts with `run-s lint:*`. A bit confusing I know 😅

File: .eslintignore [hunk 6]
```diff
@@ -1,9 +1,7 @@
-node_modules
 .loom
-.vscode-test
+.next
+build
+build-internal
+data
```
Comment (alex-page) [inline .eslintignore]: `/stylelint-polaris/data` unsure about this. Should we rename these dirs in a seperate PR

Comment (sam-b-rose) [reply .eslintignore]: Good call, will update to put these generated data items in a tmp folder instead. I feel that is a more standard practice and naming convention

Comment (alex-page) [reply .eslintignore]: tmp-tokens even might be a good call to clarify they will be removed with tokens

File: .eslintignore [hunk 7]
```diff
@@ -1,9 +1,7 @@
-node_modules
 .loom
-.vscode-test
+.next
+build
+build-internal
```
Comment (alex-page) [inline .eslintignore]: I would prefer to keep these opinionated for now
```suggestion
/polaris-react/build
/polaris-react/build-internal
```

File: .eslintignore [hunk 8]
```diff
@@ -1,9 +1,7 @@
...
 dist
-out
-/polaris-react/build
-/polaris-react/build-internal
-/polaris.shopify.com/.next
-/stylelint-polaris/data
+node_modules
```
Comment (alex-page) [inline .eslintignore:6]:
```suggestion
```

File: .gitignore [hunk 9]
```diff
@@ -1,6 +1,13 @@
...
 *.tsbuildinfo
-node_modules
+*.vsix
 .DS_Store
+.eslintcache
 .loom
+.next
 .shadowenv.d
+build
+build-internal
```
Comment (alex-page) [inline .gitignore]: These are already in a `.gitignore` in `polaris-react` so maybe we can remove it?

Comment (alex-page) [reply .gitignore]: `*.scss.d.ts` could be added globally and removed from `polaris-react` `.gitignore`.

File: .eslintignore (hunk 7 above)
Comment (sam-b-rose) [reply .eslintignore]: Agreed, updated in other `.*ignore` files as well

File: .gitignore (hunk 9 above)
Comment (sam-b-rose) [reply .gitignore]: I removed all other `.gitignore` files in favor of having a single ignore file

Comment (sam-b-rose) [reply .gitignore]: > `*.scss.d.ts` could be added globally

Do we need to add it if it is already part of the build output? Just trying to keep the ignores trim

File: .eslintignore (hunk 8 above)
[the remark this answers is not in the document]
Comment (samrose3) [reply .eslintignore:6]: This was just me keeping these files alphabetical. Very superfluous and not documented, but I felt it a reasonable cleanup given while modifying these ignore files

**Review by alex-page — APPROVED** [review]

**Review by alex-page — APPROVED** [review]

**Comment by tmlayton** [comment]
I see the `build-consumer` script was removed. There is still documentation in the README that refers to this script as the way to test in consuming apps. Also, the script itself still exists.

**Pull request merged**